internal/middleware/user.go

```diff
@@ -43,17 +47,23 @@ func (u *UserSetter) setUser(ctx *gin.Context) {
 	if username == "" {
 		panic("username header unset")
 	}
-	if user, existsInCache := u.cache.Get(username); existsInCache {
-		ctx.Set(userContextKey, user)
-		return
-	}
```

Comment:
the cache get I think needs to remain before the db read...

Reply:
confused because if I move the cache before persistedUser() then it complains that there's no user 😅

Reply:
I mean like move 
```go
	if user, existsInCache := u.cache.Get(username); existsInCache {
		ctx.Set(userContextKey, user)
		return
	}
```
back here, and leave
```go
	if UserIsCacheable(user) {
		_ = u.cache.Add(username, user)
	}
```
below

Reply:
ah, I was reading it as check if the user is cachable before checking in the cache for it because if it's not cachable, it won't be there anyway